Jenkinsfile_updatecli

```diff
@@ -0,0 +1,58 @@
+def cronExpr = env.BRANCH_IS_PRIMARY ? 'H/30 * * * *' : ''
+
+pipeline {
+  agent { label 'linux-amd64-docker' }
+
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    timeout(time: 30, unit: 'MINUTES')
+    disableConcurrentBuilds()
+  }
+
+  triggers {
+    cron (cronExpr)
+  }
+
+  environment {
+    AWS_ACCESS_KEY_ID     = credentials('packer-aws-access-key-id')
+    AWS_SECRET_ACCESS_KEY = credentials('packer-aws-secret-access-key')
+    AWS_DEFAULT_REGION    = 'us-east-2'
+  }
+  stages {
+    stage('Check Configuration Update') {
+      // Run updatecli's diff on both push and pull requests (in case a configuration change breaks updatecli)
+      steps {
+        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+          withCredentials([
+            usernamePassword(
+            credentialsId: 'github-app-updatecli-on-jenkins-infra',
+            usernameVariable: 'USERNAME_VALUE', // Setting this variable is mandatory, even if of not used when the credentials is a githubApp one
+            passwordVariable: 'UPDATECLI_GITHUB_TOKEN'
+            )
+          ]) {
+            sh 'updatecli version'
+            sh 'updatecli diff --config ./updatecli/weekly.d --values ./updatecli/values.yaml'
+          }
+        }
+      }
+    } // stage
+    stage('Apply Configuration Update') {
+      when {
+        expression { env.BRANCH_IS_PRIMARY }
+      }
+      steps {
+        catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
```

Comment:
```suggestion
```

Blocking: if the "updatecli apply" step fails, then the pipeline must fail. The `catchError()` is only for `updatecli diff` to handle cases where a new code needs to be merged on the principal branch before updatecli can handle it.